akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/RotatingKeysSSLEngineProviderSpec.scala

```diff
@@ -269,13 +268,13 @@ abstract class RotatingKeysSSLEngineProviderSpec(extraConfig: String)
   }
 
   override def beforeTermination(): Unit = {
-    systemsToTerminate.foreach { systemToTerminate =>
-      system.log.info(s"Terminating $systemToTerminate...")
-      val now = System.nanoTime()
-      Await.result(systemToTerminate.terminate(), 8.seconds.dilated)
-      val lasted = System.nanoTime() - now
-      system.log.info(s"Terminated $systemToTerminate after ${lasted / 1000000} ms")
-    }
+    systemsToTerminate.map { systemToTerminate =>
+    val patienceConfig1 = implicitly[PatienceConfig]
+      system.log.info(s"Terminating $systemToTerminate... $patienceConfig1")
+              systemToTerminate.terminate()
+    }.foreach(fut => eventually(fut))
```

Comment:
could also use the `shutdown` utility from TestKit, which gives some useful information if shutdown can't be completed
```
    systemsToTerminate.foreach(_.terminate())
    systemsToTerminate.foreach {
      shutdown(_, duration = 10.seconds, verifySystemShutdown = true)
    }
```

Reply:
TIL